include/bluetooth/hci_vs.h

```diff
@@ -163,6 +164,142 @@ struct bt_hci_evt_vs_scan_req_rx {
 #define BT_EVT_MASK_VS_TRACE_INFO              BT_EVT_BIT(2)
 #define BT_EVT_MASK_VS_SCAN_REQ_RX             BT_EVT_BIT(3)
 
+/* Mesh HCI commands */
+#define BT_HCI_MESH_REVISION                   0x01
+
+#define BT_HCI_OP_VS_MESH                      BT_OP(BT_OGF_VS, 0x0042)
+#define BT_HCI_MESH_EVT_PREFIX                 0xF0
+
+struct bt_hci_cp_mesh {
+	u8_t         opcode;
+} __packed;
+
+#define BT_HCI_OC_MESH_GET_OPTS                0x00
+struct bt_hci_rp_mesh_get_opts {
+	u8_t      status;
+	u8_t      opcode;
+	u8_t      revision;
+	u8_t      ch_map;
+	s8_t      min_tx_power;
+	s8_t      max_tx_power;
+	u8_t      max_scan_filter;
+	u8_t      max_filter_pattern;
+	u8_t      max_adv_slot;
+	u8_t      max_tx_window;
+	u8_t      evt_prefix_len;
+	u8_t      evt_prefix;
+} __packed;
+
+#define BT_HCI_MESH_PATTERN_LEN_MAX            0x0f
+
+#define BT_HCI_OC_MESH_SET_SCAN_FILTER         0x01
+struct bt_hci_mesh_pattern {
+	u8_t pattern_len;
+	u8_t pattern[0];
+} __packed;
+
+struct bt_hci_cp_mesh_set_scan_filter {
+	u8_t      scan_filter;
+	u8_t      filter_dup;
+	u8_t      num_patterns;
+	struct    bt_hci_mesh_pattern patterns[0];
```

Comment:
@carlescufi This does not translate to Pattern_Length[i], Pattern[i] in the Spec document.

Reply:
@cvinayak I will  fix this today

Reply:
As checked with @holtmann, this is correct